### DISCUSSION TITLE
where can the beginner set the value of ${tycho.version}?

### DISCUSSION THREAD

**Discussion by zhiyoutech** [eclipse-pde/eclipse.pde#655] [Q&A]
Dear,
       i am a beginner in eclipse-pde，the default value of ${tycho.version} is 4.0.0, how can i reset a new value for it?

**Comment by laeubi** [comment]
If you refer to the commandlinebuild you can always pass `-Dtycho.version=<desired version>`, nerveless it does not seems very useful do to so, so you probably want to explain more what you want to archive or what problem needs to be solved.

**Reply by zhiyoutech** [reply]
I rebuild eclipse-pde with maven (version is 3.9.2) in eclipse 2023-03, there is an error :

[ERROR] Plugin org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.0 or one of its dependencies could not be resolved: The following artifacts could not be resolved: org.eclipse.tycho.extras:tycho-p2-extras-plugin:jar:4.0.0 (absent): org.eclipse.tycho.extras:tycho-p2-extras-plugin:jar:4.0.0 was not found in https://repo.eclipse.org/content/repositories/tycho-snapshots/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of tycho-snapshots has elapsed or updates are forced -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -X  switch.
[ERROR] Re-run Maven using the -X  switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException

how could I solve this problem?
thanks!

**Reply by laeubi** [reply]
> how could I solve this problem?

Please try to pass `-U` to maven, the message indicates it has an old cached attempt that failed so it does not find the version on central.

**Reply by zhiyoutech** [reply]
it does not work. the same problem appear again.
the version  of JavaSE in my windows 10 system is JavaSE-19.

**Reply by laeubi** [reply]
Are you using a firewall or special maven settings?

**Reply by zhiyoutech** [reply]
only one compoment in tycho is not avaliable, not all compoments.
so i think it is not caused by firewall or special maven settings.

**Reply by laeubi** [reply]
The "missing" artifact is here: 

https://mvnrepository.com/artifact/org.eclipse.tycho.extras/tycho-p2-extras-plugin/4.0.0

what you can try is to start with a fresh maven repository to make sure there are no left overs.
